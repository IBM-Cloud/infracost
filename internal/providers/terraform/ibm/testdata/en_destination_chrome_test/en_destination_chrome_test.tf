terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}

locals {
  event_notifications = {
    plans                = ["lite", "standard"],
    destination_pre_prod = ["true", "false"]
  }
}

resource "ibm_resource_instance" "event_notifications" {
  for_each          = toset(local.event_notifications.plans)
  name              = "event-notifications-${each.value}"
  location          = "us-south"
  plan              = each.value
  resource_group_id = "default"
  service           = "event-notifications"
}

resource "ibm_en_destination_chrome" "destination_chrome_lite" {
  for_each              = toset(local.event_notifications.destination_pre_prod)
  instance_guid         = ibm_resource_instance.event_notifications["lite"].guid
  name                  = "Chrome Destination"
  type                  = "push_chrome"
  collect_failed_events = false
  description           = "Chrome push destination"
  config {
    params {
      api_key     = "apikey"
      website_url = "https://testevents.com"
      pre_prod    = tobool(each.value)
    }
  }
}

resource "ibm_en_destination_chrome" "destination_chrome_standard" {
  for_each              = toset(local.event_notifications.destination_pre_prod)
  instance_guid         = ibm_resource_instance.event_notifications["standard"].guid
  name                  = "Chrome Destination"
  type                  = "push_chrome"
  collect_failed_events = false
  description           = "Chrome push destination"
  config {
    params {
      api_key     = "apikey"
      website_url = "https://testevents.com"
      pre_prod    = tobool(each.value)
    }
  }
}

