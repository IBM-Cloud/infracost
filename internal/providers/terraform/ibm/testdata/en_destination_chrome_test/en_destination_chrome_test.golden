
 Name                                                                  Monthly Qty  Unit                    Monthly Cost 
                                                                                                                         
 ibm_en_destination_chrome.destination_chrome_lite["false"]                                                              
 └─ Push Destination Instances (Lite plan)                                       1  Instance                       $0.00 
                                                                                                                         
 ibm_en_destination_chrome.destination_chrome_lite["true"]                                                               
 └─ Pre-Prod Push Destination Instances (Lite plan)                              1  Instance                       $0.00 
                                                                                                                         
 ibm_en_destination_chrome.destination_chrome_standard["false"]                                                          
 └─ Push Destination Instances (Standard plan)                                   1  Instance                      $35.00 
                                                                                                                         
 ibm_en_destination_chrome.destination_chrome_standard["true"]                                                           
 └─ Pre-Prod Push Destination Instances (Standard plan)                          1  Instance                      $10.00 
                                                                                                                         
 ibm_resource_instance.event_notifications["lite"]                                                                       
 └─ Lite Plan                                                                    1  Instance                       $0.00 
                                                                                                                         
 ibm_resource_instance.event_notifications["standard"]                                                                   
 └─ Ingested Events                                              Monthly cost depends on usage: $1.08 per Million Events 
                                                                                                                         
 OVERALL TOTAL                                                                                                    $45.00 
──────────────────────────────────
6 cloud resources were detected:
∙ 6 were estimated, 2 of which include usage-based costs, see https://infracost.io/usage-file